### PR TITLE
add package units

### PR DESCRIPTION
**Pull request by healther**

### CODE REVIEW

File: var/spack/repos/builtin/packages/units/package.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+class Units(AutotoolsPackage):
+    """GNU units converts between different systems of units"""
+
+    homepage = "https://www.gnu.org/software/units/"
+    url      = "https://ftp.gnu.org/gnu/units/units-2.13.tar.gz"
+
+    version('2.13', '5cbf2a6af76e94ba0ac55fc8d99d5a3e')
+
+    depends_on('python')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/units/package.py]: is it `build`, `link` and/or `run` type dependency?

**Comment by healther** [comment]
It builds without the explicit dependency... But I guess it can just take the system python, is there a way to test that? I.e. deactivating the system's python for the build/install?

**Comment by healther** [comment]
Btw. could you just have merged only one commit? Or does github only allow the complete merge of a PR?

**Comment by adamjstewart** [comment]
> It builds without the explicit dependency... But I guess it can just take the system python, is there a way to test that? I.e. deactivating the system's python for the build/install?

Only if:

1. Python is installed on your system
2. The right version of Python is installed on your system

We generally always add dependencies. It gives the user more control over how their software is built and prevents build failures when they happen to be on a weird bare-bones system without a lot of software installed. I can't imagine a system without some version of Python installed, but I do know some newer ones only come with Python 3, which isn't always tested by developers.

**Comment by adamjstewart** [comment]
> Btw. could you just have merged only one commit? Or does github only allow the complete merge of a PR?

As far as I know, it only allows a complete merge.

**Comment by healther** [comment]
>  I can't imagine a system without some version of Python installed, but I do know some newer ones only come with Python 3, which isn't always tested by developers.

I know, I had hoped for a spack-feature that allows me to prevent setup/configure/make routines from seeing the system at all and only relying on spack-provided binaries. With this I would have seen the missing dependency, which was hidden by me having a fitting python version.

**Comment by healther** [comment]
The question is whether this "breaking things very badly" is a sign of general misunderstanding/ lacking knowledge of ones dependencies -> I'm not sure whether it wouldn't be better to just break things instead of "works for me but not for you"-situations.

**Comment by adamjstewart** [comment]
@healther But then we would have to create a package for literally everything in `/bin` and `/usr/bin`. Everything with a Makefile would have to depend on `make`, etc.

**Pull request merged**